src/zope/server/ftp/server.py

```diff
@@ -837,41 +838,22 @@ def reportDefault(self):
 
 
 @implementer(ITask)
-class FinishSTORTask(object):
+class FinishSTORTask(AbstractTask):
     """Calls control_channel.finishSTOR() in an application thread.
 
     This task executes after the client has finished uploading.
     """
 
     def __init__(self, control_channel, inbuf, finish_args):
-        self.control_channel = control_channel
+        AbstractTask.__init__(self, control_channel)
         self.inbuf = inbuf
         self.finish_args = finish_args
 
-    def service(self):
-        """Called to execute the task.
-        """
-        close_on_finish = 0
-        c = self.control_channel
-        try:
-            try:
-                c.finishSTOR(self.inbuf, self.finish_args)
-            except socket.error:
-                close_on_finish = 1
-                if c.adj.log_socket_errors:
-                    raise
-        finally:
-            if close_on_finish:
-                c.close_when_done()
-
-    def cancel(self):
-        'See ITask'
-        self.control_channel.close_when_done()
-
-    def defer(self):
-        'See ITask'
-        pass
+    def _do_service(self):
+        self.channel.finishSTOR(self.inbuf, self.finish_args)
 
+    def finish(self):
+        "Does nothing"
```

Comment:
Tripple `"""` for docstrings please?

Reply:
Sure, will do (everywhere you caught me!). Some of these are mine and some are existing code that was moved around. This package isn't terrible consistent when it comes to that.